src/main/org/codehaus/groovy/runtime/ConversionHandler.java

```diff
@@ -189,4 +202,15 @@ public static boolean isCoreObjectMethod(Method method) {
         return Object.class.equals(method.getDeclaringClass());
     }
 
+    private MetaClass setMetaClass(MetaClass mc) {
+        metaClass = mc;
+        return mc;
+    }
+
+    private MetaClass getMetaClass(Object proxy) {
+        if (metaClass == null) {
+            metaClass = ((MetaClassRegistryImpl) GroovySystem.getMetaClassRegistry()).getMetaClass(proxy);

```

Comment:
Not sure if this is the right way to get the `metaClass` for the `Proxy` object.